### PR TITLE
fix(middleware): respect host protocol

### PR DESCRIPTION
**Pull request by vicb**
The middleware handler default to https and it's causing troubles with the cloudflare dev server (wrangler dev) which uses http by default.

It is possible to force wrangler dev to use https (--local-protocol https) but the browser then shows a warning because the dev certificate is not secure.

### CODE REVIEW

**Comment by vicb** [comment]
@conico974 I have only update the middleware as I'm not sure if we will use the edge-adapter.

File: packages/open-next/src/core/routing/middleware.ts [hunk 1]
```diff
@@ -54,8 +54,10 @@ export async function handleMiddleware(
   // We bypass the middleware if the request is internal
   if (internalEvent.headers["x-isr"]) return internalEvent;
 
+  const protocol = new URL(internalEvent.url).protocol;
```
Comment (conico974) [inline packages/open-next/src/core/routing/middleware.ts:57]: We can't do that here, in lambda `url` only contains the rawPath and the query, no protocol or host
We could use https://github.com/opennextjs/opennextjs-aws/blob/811bdc0008213eed9f859fc505c92d85327a1bac/packages/open-next/src/core/routing/util.ts#L44 along with isExternal https://github.com/opennextjs/opennextjs-aws/blob/811bdc0008213eed9f859fc505c92d85327a1bac/packages/open-next/src/core/routing/util.ts#L19

If not found we default to `https`, would this work for you ?

**Review by conico974 — COMMENTED** [review]
What's the issue with the edge-adapter ?

**Comment by vicb** [comment]
@conico974 PTAL and let me know if the implementation is ok. Thanks!

**Review by conico974 — APPROVED** [review]
LGTM Thanks

**Pull request merged**